web/html/src/manager/maintenance/maintenance-windows-details.js

```diff
@@ -38,59 +38,103 @@ class MaintenanceWindowsDetails extends React.Component {
-                    <table className="table">
-                        <tbody>
-                        <tr>
-                            <td>{t("Schedule Name")}</td>
-                            <td>{data.scheduleName}</td>
-                        </tr>
-                        <tr>
-                            <td>{t("Assigned Calendar")}:</td>
-                            <td>{data.calendarName}</td>
-                        </tr>
-                        <tr>
-                            <td>{t("Schedule Type")}:</td>
-                            <td>{data.scheduleType === "SINGLE" ? t("Single") : t("Multi")}</td>
-                        </tr>
-                        </tbody>
-                    </table>
-                </div>
-                <DeleteDialog id="delete-modal"
-                              title={t("Delete maintenance schedule")}
-                              content={t("Are you sure you want to delete the selected item? \n" +
-                                  "This will remove the current schedule from all the systems assigned to it.")}
-                              onConfirm={() => this.props.onDelete(this.props.data)}
-                />
-            </BootstrapPanel>
+            <div>
+                <BootstrapPanel title={t("Schedule Details")}>
+                    <div className="table-responsive">
+                        <table className="table">
+                            <tbody>
+                            <tr>
+                                <td>{t("Schedule Name")}</td>
+                                <td>{data.scheduleName}</td>
+                            </tr>
+                            <tr>
+                                <td>{t("Assigned Calendar")}:</td>
+                                <td>{data.calendarName}</td>
+                            </tr>
+                            <tr>
+                                <td>{t("Schedule Type")}:</td>
+                                <td>{data.scheduleType === "SINGLE" ? t("Single") : t("Multi")}</td>
+                            </tr>
+                            </tbody>
+                        </table>
+                    </div>
+                    <DeleteDialog id="delete-modal"
+                                  title={t("Delete maintenance schedule")}
+                                  content={t("Are you sure you want to delete the selected item? \n" +
+                                      "This will remove the current schedule from all the systems assigned to it.")}
+                                  onConfirm={() => this.props.onDelete(this.props.data)}
+                    />
+                </BootstrapPanel>
+                {
+                    data.maintenanceWindows !== undefined && data.maintenanceWindows.length > 0 &&
+                    <BootstrapPanel title={t("Upcoming Maintenance Windows")}>
+                        <div className="table-responsive">
+                            <table className="table">
```

Comment:
Would it be possible to use some existing component for the table (`Table.js`)? Actually, in this PR i see quite some table crafting, maybe it'd make sense to re-use something either. @ncounter What's the general advice here? Re-use components (even for simple tables), or craft tables on-the-fly?

Reply:
I believe I kind of overlooked at this, I remember we discussed with @parlt91 but then we never got back on this. Probably in this case it will be a bit of overengineering work to use the existing component, but we really should. User will never need to use Table.js component feature (ordering by column, comparing values, etc) but it is important to keep every table look&feel fetching structure from the same piece of code.
@parlt91 if you need any help to adapt and make use of the existing table component, just ping me. I think we really should do this now that we have "a bit of time" :wink: 
Thanks @hustodemon for pointing it out

Reply:
Aright. I will look into this. The component seems to work without the headers it's just that there is some empty area remaining where the headers are supposed to be.
I will have to see how it looks. Maybe I need to do some adjustments to the Table but let's see.